test/test_pipeline_schedule.py

```diff
@@ -34,9 +35,42 @@
     PipelineScheduleLoopedDFS,
     PipelineStage,
 )
+from torch.profiler import record_function
 
 logger = logging.getLogger(__name__)
 
+_null_context = nullcontext()
+
+
+# profiling context manager
+@contextmanager
+def maybe_run_profiler(
+    use_profiler, trace_dir, schedule, rank, *args, **kwargs
+):
+    def trace_handler(prof):
+        if rank == 0:
+            (f"about to EXPORT traces for {schedule} to {trace_dir}")
+        prof.export_chrome_trace(
+            f"{trace_dir}/{schedule}_rank{rank}_trace.json"
+        )
+
+    if use_profiler:
+        with torch.profiler.profile(
+            activities=[
+                torch.profiler.ProfilerActivity.CPU,
+                torch.profiler.ProfilerActivity.CUDA,
+            ],
+            # schedule=torch.profiler.schedule(wait=1, warmup=2, active=3, repeat=1),
```

Comment:
nice i didnt know about schedule `arg` in profile. Looks like this could be useful in the future if we wanted to profile for a particular step or even microbatch

Reply:
yes definitely should be useful as we expand out the profiling coverage.